modules/preview-react/radio/lib/RadioInput.tsx

```diff
@@ -0,0 +1,218 @@
+    height: '18px',
+    width: '18px',
+
+    // Circle element styles the radio as checked or unchecked
+    ':after': {
+      content: "''",
+      display: 'flex',
+      flexDirection: 'column',
+      alignItems: 'center',
+      backgroundColor: disabled ? inputColors.disabled.background : colors.frenchVanilla100,
+      borderRadius: '999px',
+      boxSizing: 'border-box',
+      border: `1px solid`,
+      borderColor: disabled
+        ? inputColors.disabled.border
+        : variant === 'inverse'
+        ? colors.soap300
+        : inputColors.border,
+      height: '18px',
+      width: '18px',
+      justifyContent: 'center',
+      pointerEvents: 'none',
+      position: 'absolute',
+      transition: 'border 200ms ease, background 200ms',
+      opacity: disabled && variant === 'inverse' ? '.4' : '1',
+    },
+
+    '&:hover:after': {
+      borderColor: disabled
+        ? inputColors.disabled.border
+        : variant === 'inverse'
+        ? colors.soap300
+        : colors.licorice500,
+    },
+
+    '&:focus:after': {
+      borderColor: disabled
+        ? inputColors.disabled.border
+        : variant === 'inverse'
+        ? colors.soap300
+        : colors.blueberry400,
+    },
+
+    '&:checked:after': {
+      backgroundColor: variant === 'inverse' ? themePrimary.main : colors.frenchVanilla100,
+      border: `5px solid`,
+      borderColor: variant === 'inverse' ? colors.frenchVanilla100 : themePrimary.main,
+    },
+
+    '&:focus:after, &:focus:hover:after': {
+      outline: 'transparent',
+
+      ...focusRing({
+        width: variant === 'inverse' ? 2 : 1,
+        separation: 0,
+        animate: false,
+        innerColor: variant === 'inverse' ? colors.blackPepper400 : colors.frenchVanilla100,
+        outerColor: variant === 'inverse' ? colors.frenchVanilla100 : colors.blueberry400,
+      }),
+    },
+
+    '&:focus:checked:after, &:focus:hover:checked:after': {
+      outline: 'transparent',
+      ...focusRing({
+        width: variant === 'inverse' ? 2 : 2,
+        separation: 2,
+        animate: false,
+        innerColor: variant === 'inverse' ? colors.blackPepper400 : colors.frenchVanilla100,
+        outerColor: variant === 'inverse' ? colors.frenchVanilla100 : colors.blueberry400,
+      }),
+    },
+
+    ...mouseFocusBehavior({
+      '&:focus:after': {
+        ...focusRing({
+          width: 0,
+          outerColor: variant === 'inverse' ? colors.frenchVanilla100 : themeFocusOutline,
+        }),
+      },
+      '&:focus:hover:after, &:focus:active:after': {
+        ...focusRing({
+          width: 0,
+          outerColor: variant === 'inverse' ? colors.frenchVanilla100 : themeFocusOutline,
+        }),
+      },
+      '&:focus:hover:checked:after, &:focus:active:checked:after': {
+        ...focusRing({
+          width: 0,
+          outerColor: variant === 'inverse' ? colors.frenchVanilla100 : themeFocusOutline,
+        }),
```

Comment:
you have the same 3 focus rings, should it be united under one selector?

```
'&:focus:after, &:focus:hover:after, &:focus:active:after, &:focus:hover:checked:after, &:focus:active:checked:after' : {}
```

Reply:
yep! thank you!